hot fix for key structure

jrknezha opened the pull request:
This PR updates our ability to process data in buckets that have a different structure than /key_name/date_values. Now the user provides the full key path including the cycle date values so we can handle structures such as /bucket/%Y/%m/%Y%m%d%H/folder. 

This is a change in functionality and will require new behaviors for users by providing the full path including cycle in the .env file as well as new registrations of storage locations to use this value as the key (that process remains the same). The change in input expectation is reflected in the README changes. 

The new code has been cycled for file_counts, inc_logs, and daily_mean_surface_background to exercise all the new code and verified with successful inputs in the database. 

This is a hot fix as users need this functionality to use the code now. Fixes issue #16.

amschne commented on README.md:
```diff
@@ -43,14 +43,14 @@ EXPERIMENT_NAME = 'EXAMPLE-EXPERIMENT-NAME'
 STORAGE_LOCATION_BUCKET is the root name of the S3 bucket, this must match what is in AWS.
 STORAGE_LOCATION_PLATFORM is a metadata value used for referencing the storage location for registration and file counts.
-STORAGE_LOCATION_KEY is the key location in the S3 bucket beneath the root to be used. This will be used for metadata registration and pulling data. The value can be an empty string if the top of the S3 bucket is being used and no '/' should be in the front of the key. 
+STORAGE_LOCATION_KEY is the key location in the S3 bucket beneath the root to be used. This will be used for metadata registration and pulling data. The value can be an empty string if the top of the S3 bucket is being used and no '/' should be in the front of the key. It should include any year, month, and date string replacement values in the standard format such as %Y/%m/%Y%m%d%H.
```
```suggestion
STORAGE_LOCATION_KEY is the key location in the S3 bucket beneath the root to be used. This will be used for metadata registration and pulling data. The value can be an empty string if the top of the S3 bucket is being used and no '/' should be in the front of the key. It should include any year, month, and date string replacement values using standard format codes (e.g., %Y/%m/%Y%m%d%H for the string formatted as [YYYY]/[MM]/[YYYYMMDDHH]). Documentation for Python's datetime format codes are provided at [https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes).
```

amschne reviewed (commented):
This PR will work while improving usability for different experiment configurations. Because it changes (and improves) how users will interact with repo when configuring their particular use cases, I have some suggestions for further improvement. But I'll leave it to your discretion, @jrknezha.

jrknezha commented on scripts/db_file_count.py:
```diff
@@ -42,10 +42,8 @@
-else: 
-    prefix = year + "/" + month + "/" + datetime_str + "/"
+
+prefix = datetime_obj.strftime(key)
```
updated

jrknezha commented:
Thanks for your comments, one of your comments highlighted that I added a trailing slash but that was not how the previous design worked (and may cause issues for running at the top of the bucket), so I went back in and added the slash into the scripts the way it was before. I also updated the code based on your comments. 

I have re-tested the new code and it's all running successfully.

amschne reviewed (approved):
tested successfully and approved pending the minor possible corrections/updates to the docstrings in scripts/*.py

amschne commented on scripts/db_inc_logs.py:
I suspect this goes for all the .py scripts
